apps/mgmt-ui/src/components/configuration/IntegrationTabPanel.tsx

```
@@ -151,7 +151,7 @@ export default function IntegrationTabPanel(props: IntegrationDetailTabPanelProp
                   ei.id === newIntegration.id ? newIntegration : ei
                 );
 
-                updateRemoteIntegration(activeApplication.id, newIntegration);
+                updateRemoteIntegration(activeApplicationId as string, newIntegration);
```

is there a way to avoid doing this everywhere?

---

Not sure as this is at the component level, and there is no guarantee that the router has `applicationId` set. We can throw an exception I guess, but is that really what we want?

---

I'm throwing an exception